### PR TITLE
Update doc testsuite to use 1.9.6

### PR DESCRIPTION
**Pull request by sloriot**
Replaces #6178

### CODE REVIEW

File: Documentation/doc/resources/1.9.6/BaseDoxyfile.in [hunk 1]
```diff
@@ -0,0 +1,812 @@
...
+# Note: The order of the extra style sheet files is of importance (e.g. the last
+# style sheet in the list overrules the setting of the previous ones in the
+# list).
+# Note: Since the styling of scrollbars can currently not be overruled in
+# Webkit/Chromium, the styling will be left out of the default doxygen.css if
+# one or more extra stylesheets have been specified. So if scrollbar
+# customization is desired it has to be added explicitly. For an example see the
+# documentation.
+
+HTML_STYLESHEET        = ${CGAL_DOC_RESOURCE_DIR}/stylesheet.css
```
Comment (albert-github) [inline Documentation/doc/resources/1.9.6/BaseDoxyfile.in:488]: Strange documentation for `HTML_EXTRA_STYLESHEET` and setting is called `HTML_STYLESHEET`

**Comment by albert-github** [comment]
Regarding the part in the Documentation/doc and resource/1.9.6 directory why not use here the parts from the proposed pull request #7427?  With the following actions:
- rename there the directory from 1.9.7 to 1.9.6
- place the 1.9.6 instead of 1.9.7 in the CMakeList.txt
- update the BaseDoxyfile.in so it is 1.9.6 conforming (remove `MARKDOWN_ID_STYLE` and `MSCGEN_TOOL` parts)

in this (#7441) PR I also miss the usage of the updated extra stylesheet cgal_stylesheet.css plus the default doxygen.css stylesheet in favor of a updated copy of doxygen.css (and a renaming to cgal_stylesheet.css).
(Also looks like not correct in the BaseDoxyfile.in, see the comment with the code. I didn't review the code further)

Maybe an updated version of #6178 should be added to #7427 to get some postprocessing / internal CGAL tests right

File: Documentation/doc/resources/1.9.6/BaseDoxyfile.in (hunk 1 above)
Comment (sloriot) [reply Documentation/doc/resources/1.9.6/BaseDoxyfile.in:488]: That's from `doxygen -g`

**Comment by sloriot** [comment]
> Regarding the part in the Documentation/doc and resource/1.9.6 directory why not use here the parts from the proposed pull request #7427? With the following actions:
> 
>     * rename there the directory from 1.9.7 to 1.9.6
> 
>     * place the 1.9.6 instead of 1.9.7 in the CMakeList.txt
> 
>     * update the BaseDoxyfile.in so it is 1.9.6 conforming (remove `MARKDOWN_ID_STYLE` and `MSCGEN_TOOL` parts)
> 
> 
> in this (#7441) PR I also miss the usage of the updated extra stylesheet cgal_stylesheet.css plus the default doxygen.css stylesheet in favor of a updated copy of doxygen.css (and a renaming to cgal_stylesheet.css). (Also looks like not correct in the BaseDoxyfile.in, see the comment with the code. I didn't review the code further)
> 
> Maybe an updated version of #6178 should be added to #7427 to get some postprocessing / internal CGAL tests right

That was my initial goal but then I was affected by several issues so I decided to go the easy way. Your changes will be tested in the test of the master version of doxygen.

**Comment by albert-github** [comment]
Major worry I have is when will my changes be incorporated into the production version of CGAL (as normally it takes a while before an update occurs).

File: Documentation/doc/resources/1.9.6/BaseDoxyfile.in (hunk 1 above)
Comment (albert-github) [reply Documentation/doc/resources/1.9.6/BaseDoxyfile.in:488]: That is very strange, I have not been able to reproduce this with either `doxygen -g` nor with doxygen -u` 
(With the `doxygen -u` I get some warnings about an environment variable that replaces a complete settings line:
```
warning: ignoring unknown character '$' at line 610, file BaseDoxyfile.in
warning: ignoring unknown character '{' at line 610, file BaseDoxyfile.in
warning: ignoring unknown tag 'CGAL_OPT_EXTRACT_ALL_NO_DETAILED_IF_EMPTY' at line 610, file BaseDoxyfile.in
warning: ignoring unknown character '}' at line 610, file BaseDoxyfile.in
warning: ignoring unknown character '$' at line 1717, file BaseDoxyfile.in
warning: ignoring unknown character '{' at line 1717, file BaseDoxyfile.in
warning: ignoring unknown tag 'CGAL_DOC_MATHJAX_LOCATION_FULL_OPTION_LINE' at line 1717, file BaseDoxyfile.in
warning: ignoring unknown character '}' at line 1717, file BaseDoxyfile.in
```

Comment (sloriot) [reply Documentation/doc/resources/1.9.6/BaseDoxyfile.in:488]: I just checked and it is there:
```
doxygen --version
1.9.6 (523d3509b8630cf9ee508f9196ff38555a8e18a4*)
```
The branch is based on
```
commit 4586b5cfaa3d46d51f6a51882951d15644c49edf (tag: Release_1_9_6)
Author: Dimitri van Heesch <doxygen@gmail.com>
Date:   Mon Dec 26 20:53:19 2022 +0100

    Update changelog for release 1.9.6
```

**Comment by sloriot** [comment]
> Major worry I have is when will my changes be incorporated into the production version of CGAL (as normally it takes a while before an update occurs).

I tested my changes and the test machine generated everything correctly. I removed everything to let the testsuite run automatically tonight. So stating tomorrow I can start adding your branch for testing.

File: Documentation/doc/resources/1.9.6/BaseDoxyfile.in (hunk 1 above)
Comment (sloriot) [reply Documentation/doc/resources/1.9.6/BaseDoxyfile.in:488]: -g

Comment (albert-github) [reply Documentation/doc/resources/1.9.6/BaseDoxyfile.in:488]: But how did the things like `${CGAL_DOC_RESOURCE_DIR}/stylesheet.css` land in the file?
`doxygen -g` will just make a default `Doxyfile`.

Comment (sloriot) [reply Documentation/doc/resources/1.9.6/BaseDoxyfile.in]: I misread you first comment, the doc is indeed not the right one,

Comment (albert-github) [reply Documentation/doc/resources/1.9.6/BaseDoxyfile.in]: Looks better now, though `HTML_EXTRA_STYLESHEET` (although empty) is still missing, so it is not 100% based on a 1.9.6 Doxyfile.

**Comment by albert-github** [comment]
I just tested (against the CGAL master version) the results of this PR (and as partially indicated before) there are some big problems
- search in not working with the different packages only with Manual package it is present
- the "magnifying glass" is missing for the Manual package
- the divider between the treeview and the content is not present (probably a problem with the footer)

Note:
- doxygen 1.9.7 has been released so I think it would certainly be good to quickly switch to the proposed pull request https://github.com/CGAL/cgal/pull/7427 and adjust the testing procedure.

**Comment by lrineau** [comment]
What do we plan to ship with CGAL-5.6? Still the doc produced by Doxygen version 1.8.13?

**Comment by albert-github** [comment]
Maybe an unexpected answer from my side:
- I would take the doxygen 1.8.13 version for the CGAL release 5.6 
- switch to doxygen **1.9.7** for the CGAL 6.0 release

small reasoning:
- consistency with previous 5.* releases
- not switching directly again from the 1.9.6 to the 1.9.7 release (though I would advocate to keep a bit more in line with the doxygen releases in the future).

**Pull request merged**